packages/util/src/indexeddb.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export class IndexedDbDatabaseService {
+  dbPromise: Promise<IDBDatabase>;
+  constructor(
+    public dbName: string,
+    public storeName: string,
+    public dbVersion: number,
+    errorHandler: (error: Error) => void
+  ) {
+    this.dbPromise = new Promise((resolve, reject) => {
+      try {
+        const request = indexedDB.open(this.dbName, this.dbVersion);
+
+        request.onsuccess = event => {
+          resolve((event.target as IDBOpenDBRequest).result);
+        };
+
+        request.onerror = event => {
+          reject((event.target as IDBRequest).error?.message);
+        };
+
+        request.onupgradeneeded = event => {
```

Comment:
You'd want this callback to be provided by the users of this utility class because the upgrade logic varies from use case to use case, and thus can't be generalized.

Reply:
Removed in favor of using `idb`.